src/app/modules/widgets/tree-viewer/input-box/input-box.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, ViewChild, Input, ElementRef, AfterViewInit, Output, EventEmitter } from '@angular/core';
+import { TreeNode } from '../tree-node';
+import { InputBoxConfig, TreeViewerInputBoxConfig } from '../../../event-types-in';
+
+export enum ContextType { Parent, Sibling }
+export interface NodeContext { node: TreeNode; type: ContextType; }
+
+@Component({
+  selector: 'app-input-box',
```

Comment:
seems a bit to generic, app-tree-input-box might be a better choice (depending on this selectors scope)